### PR TITLE
feat: improve Slack bot UX with robust parsing and progressive help

### PR DESCRIPTION
**Pull request by vimalk78**
- Replace monolithic help with categorized progressive help system
  - Add dedicated help functions for each command category
  - Handle trailing commas/spaces in ParseImageInput without errors
  - Add fuzzy matching with "Did you mean?" suggestions for typos
  - Add comprehensive test coverage for parsing edge cases

### CODE REVIEW

File: pkg/slack/events/messages/message_handler.go [hunk 1]
```diff
@@ -97,59 +103,428 @@ func postResponse(client *slack.Client, event *slackevents.MessageEvent, respons
...
+	helpMessage += "• `help launch` - Launch OpenShift clusters\n"
+	helpMessage += "• `help rosa` - ROSA (Red Hat OpenShift Service on AWS)\n"
+	helpMessage += "• `list` - See active clusters\n"
+	helpMessage += "• `done` - Terminate your cluster\n"
+	helpMessage += "• `auth` - Get cluster credentials\n\n"
+
+	// All available commands with detailed usage
+	helpMessage += "\n*All Commands:*\n"
+
+	helpMessage += "\n**Cluster Launching:**\n"
```
Comment (bradmwilliams) [inline pkg/slack/events/messages/message_handler.go:121]: These double stars `**` are not translating into anything valid in slack:
image

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: pkg/slack/events/messages/message_handler.go (hunk 1 above)
Comment (vimalk78) [reply pkg/slack/events/messages/message_handler.go]: pushed new changes
